src/tools/build_helper/src/git.rs

```diff
@@ -107,6 +109,38 @@ pub fn get_git_merge_base(
     Ok(output_result(git.arg("merge-base").arg(&updated_master).arg("HEAD"))?.trim().to_owned())
 }
 
+/// Searches for the nearest merge commit in the repository.
```

Comment:
```suggestion
/// Searches for the nearest merge commit in the repository that also exists upstream.
```

Reply:
I am fine with adding that comment, but in rare cases it's not correct (like you mentioned at https://github.com/rust-lang/rust/pull/130161#issuecomment-2340775588).

Reply:
Even in those cases, the merge commit exists upstream. It's just not the one we wanted...